cmd/argocd-application-controller/commands/argocd_application_controller.go

```diff
@@ -212,10 +213,9 @@ func NewCommand() *cobra.Command {
 	return &command
 }
 
-func getClusterFilter(kubeClient *kubernetes.Clientset, settingsMgr *settings.SettingsManager, shardingAlgorithm string, enableDynamicClusterDistribution bool) sharding.ClusterFilterFunction {
-
-	var replicas int
-	shard := env.ParseNumFromEnv(common.EnvControllerShard, -1, -math.MaxInt32, math.MaxInt32)
+func getClusterSharding(kubeClient *kubernetes.Clientset, settingsMgr *settings.SettingsManager, shardingAlgorithm string, enableDynamicClusterDistribution bool) sharding.ClusterSharding {
+	var replicasCount int
+	shardNumber := env.ParseNumFromEnv(common.EnvControllerShard, 0, -math.MaxInt32, math.MaxInt32)
```

Comment:
Now that we have Deployments added, set the default value to -1.

Reply:
Can you explain that?

Reply:
So with Deployments, as we use ConfigMap to compute the ShardNumber, we can have only one Shard with id `0`, which means, multiple controller instances cannot use shard 0 for processing the applications. If we default the value to `0` here, it will end up setting the value to 0 always.

Reply:
OK, got it
So, I have fixed that by having 2 different default values depending on if we are using `sts` or `deployment`

https://github.com/argoproj/argo-cd/pull/15237/commits/bea9771c592ad2752d79f6c79de8bba442eaaf21#diff-21d1e1ae7561765d523f51ae68a0b94580d1a5980378aa8ccbaa2a7581fc0643R241